spec/02-integration/02-cmd/02-start_stop_spec.lua

```diff
@@ -588,18 +588,26 @@ describe("kong start/stop #" .. strategy, function()
 
     it("ensures the required shared dictionaries are defined", function()
       local templ_fixture     = "spec/fixtures/custom_nginx.template"
-      local new_templ_fixture = "spec/fixtures/custom_nginx.template.tmp"
+      local tmp_templ_fixture = "spec/fixtures/custom_nginx.template.tmp"
+
+      -- load the config
+      local prefix_handler = require "kong.cmd.utils.prefix_handler"
```

Comment:
From the usage here we can find that now the `custom_nginx.template` has a dependency on the common one. This counts as a reason I support the one unified template.

Reply:
> could we use a unified template instead of with a separate file called "custom_nginx.template"

we could, and I agree it would be even more clean. Not all tests can use the content of "custom_nginx.template" though, some will fail if they do, so we'd have to provide an additional variable to `helpers.start_kong` that controls if the template will be rendered as "custom" or not, and then update all integration tests accordingly. I guess I was trying to minimize the impact of this PR, maybe that should be a next step in a separate one wdyt?

Reply:
> could we use a unified template instead of with a separate file called "custom_nginx.template"

`custom_nginx.template` is a custom nginx configuration template, which includes "partial" configs such as `nginx_kong.lua` (the http config), and `nginx_kong_stream.lua` (stream conf), the rest are custom things we only need for *some* tests, which I think belong in a custom template.

Currently in this PR I am removing duplication by referring to the "partial" configs from inside the nginx configuration template, which is why now there is a dependency (it removes duplication by grabbing config from the other files).

The other default nginx configuration file where we could think of unifying things is `kong/templates/nginx.lua`, but perhaps that should be a next step because it would involve updating basically all tests and I would like to avoid making this PR grow larger, wdyt?

Reply:
@samugi I'm OK with it. Let's make it in another PR.